### PR TITLE
[develop] Change default TreeWidth value

### PR DESCRIPTION
**Pull request by NSsirena**
### Description of changes
* Change default value to TreeWidth
 
As a result of fanout investigation we decided to set the default TreeWidth value to 30 since it is the best balance between performances and robustness

### Tests
* Tests were made during the fanout investigation

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cookbooks/aws-parallelcluster-slurm/templates/default/slurm/slurm.conf.erb [hunk 1]
```diff
@@ -27,7 +27,7 @@ LaunchParameters=enable_nss_slurm
 # CLOUD CONFIGS OPTIONS
 <% if node['cluster']['use_private_hostname'] == 'true' or node['cluster']['dns_domain'].nil? or node['cluster']['dns_domain'].empty? -%>
 SlurmctldParameters=idle_on_node_suspend,power_save_min_interval=30,node_reg_mem_percent=<%= node['cluster']['slurm_node_reg_mem_percent'] %>
-TreeWidth=65533
+TreeWidth=30
```
Comment (jdeamicis) [inline cookbooks/aws-parallelcluster-slurm/templates/default/slurm/slurm.conf.erb]: This is not where the default TreeWidth is set. This is a special case when the DNS resolution is disabled.

Please leave it as it is in that line, and add `TreeWidth=30` in the `else` branch of that if condition.

**Review by jdeamicis — CHANGES_REQUESTED** [review]

**Review by jdeamicis — DISMISSED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -29,6 +29,7 @@ This file is used to list changes made in each version of the AWS ParallelCluste
 - Set Slurm BatchStartTimeout to 3 minutes so to allow max 3 minutes Prolog execution during compute node registration.
 - Upgrade Slurm to version 23.02.1.
 - Upgrade munge to version 0.5.15.
+- Set Slurm default TreeWidth to 30
```
Comment (enrico-usai) [inline CHANGELOG.md]: minor: missing final dot
minor: use backtick for TreeWidth

Comment (enrico-usai) [reply CHANGELOG.md]: What about: "Set default value for Slurm `TreeWidth` to 30."

Comment (NSsirena) [reply CHANGELOG.md]: Thanks for spotting this.
Fixing it.

**Review by jdeamicis — APPROVED** [review]

**Pull request merged**